src/Collections/Arrays/BinarySearch.dfy

```diff
@@ -0,0 +1,136 @@
+// RUN: %dafny /compile:0 "%s"
+
+include "../../StandardLibrary.dfy"
+include "../../UInt.dfy"
+
+
+
+module BinarySearch{
+  export
+    reveals Reflexive, AntiSymmetric, Connected, TotalOrdering
+    reveals LexicographicByteSeqBelow, LexicographicByteSeqBelowAux
+    provides AboutLexicographicByteSeqBelow
+    provides StandardLibrary, UInt
+
+  import StandardLibrary
+  import opened UInt = StandardLibrary.UInt
+  import opened Wrappers = StandardLibrary.Wrappers
+  
+
+predicate Reflexive<T(!new)>(R: (T, T) -> bool) {
```

Comment:
```suggestion
  predicate Reflexive<T(!new)>(R: (T, T) -> bool) {
```

And I'd suggest similar changes throughout this file to ensure the indentation is consistent.